src/vivarium/framework/results/observation.py

```diff
@@ -140,12 +139,17 @@ def __init__(
         results_formatter: Callable[[str, pd.DataFrame], pd.DataFrame],
         to_observe: Callable[[Event], bool] = lambda event: True,
     ):
+        def _wrap_results_gatherer(
+            df: pd.DataFrame, _: tuple[str, ...] | None
+        ) -> pd.DataFrame:
+            return results_gatherer(df)
+
         super().__init__(
             name=name,
             pop_filter=pop_filter,
             when=when,
             results_initializer=self.create_empty_df,
-            results_gatherer=results_gatherer,
+            results_gatherer=_wrap_results_gatherer,  # type: ignore [arg-type]
```

Comment:
This is being ignored because this creates a results gatherer that is a `Callable[[pd.DataFrame], pd.DataFrame]` where mypy thinks the typing is either a dataframe or a groupby and then the second argument is an optional tuple. We however know that it is one case or the other and have implemented the wrapper above to handle this.

Alternatively, the options I do not like is we can have the two separate callables as type hints in the base class and then we would need to ignore arg type in the observe method where we call the results gatherer because mypy does not know if df is a pd.DataFrame or groupby object.

Reply:
What if you type-hinted your wrapped method like this:
```python
def _wrap_results_gatherer(
    df: pd.DataFrame | DataFrameGroupBy[Any], _: tuple[str, ...] | None
) -> pd.DataFrame:
    if isinstance(df, DataFrameGroupBy):
        raise TypeError(...)
    return results_gatherer(df)
```

Reply:
@albrja Did Rajan's suggestion just not work?